src/main/kotlin/no/skatteetaten/aurora/gobo/graphql/toxiproxy/ToxiProxyDataLoader.kt

```diff
@@ -0,0 +1,79 @@
+package no.skatteetaten.aurora.gobo.graphql.toxiproxy
+
+import com.expediagroup.graphql.generator.annotations.GraphQLIgnore
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.module.kotlin.convertValue
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fkorotkov.kubernetes.metadata
+import com.fkorotkov.kubernetes.newPod
+import no.skatteetaten.aurora.gobo.graphql.GoboDataLoader
+import no.skatteetaten.aurora.gobo.graphql.GoboGraphQLContext
+import no.skatteetaten.aurora.gobo.integration.mokey.ApplicationService
+import no.skatteetaten.aurora.kubernetes.KubernetesCoroutinesClient
+import org.springframework.stereotype.Component
+
+@Component
+class ToxiProxyDataLoader(
+    private val applicationService: ApplicationService,
+    private val kubernetesClient: KubernetesCoroutinesClient
+) : GoboDataLoader<ToxiProxyId, List<ToxiProxy>>() {
+
+    override suspend fun getByKeys(keys: Set<ToxiProxyId>, ctx: GoboGraphQLContext): Map<ToxiProxyId, List<ToxiProxy>> {
+        return keys.associateWith { id ->
+            val applicationDeploymentDetails = applicationService.getApplicationDeploymentDetails(ctx.token(), id.applicationDeploymentId)
+            applicationDeploymentDetails.podResources.flatMap { pod ->
+                val pods = pod.containers.mapNotNull {
+                    container ->
+                    if (container.name.endsWith("-toxiproxy-sidecar")) {
+                        pod
+                    } else {
+                        null
+                    }
+                }
+
+                pods.map {
+                    val podName = it.name
+                    val deploymentRef = applicationDeploymentDetails.applicationDeploymentCommand.applicationDeploymentRef
+                    val application = deploymentRef.application
+                    val environment = deploymentRef.environment
+                    val affiliation = id.affiliation
+
+                    val pod = newPod {
```

Comment:
`pod` er allerede i bruk, kall den noe annet. feks `podInput`?

Reply:
Har du sett, to variable med samme navn.. Renamet.